pkg/virt-handler/node-labeller/cpu_plugin.go

```diff
@@ -0,0 +1,193 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package nodelabeller
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"strings"
+)
+
+const (
+	usableNo               string = "no"
+	DefaultMinCPU                 = "Penryn"
+	nodeLabellerVolumePath        = "/var/lib/kubevirt-node-labeller/"
+)
+
+var DefaultObsoleteCPUs = map[string]bool{
+	"486":        true,
+	"pentium":    true,
+	"pentium2":   true,
+	"pentium3":   true,
+	"pentiumpro": true,
+	"coreduo":    true,
+	"n270":       true,
+	"core2duo":   true,
+	"Conroe":     true,
+	"athlon":     true,
+	"phenom":     true,
+	"kvm32":      true,
+	"kvm64":      true,
+	"qemu32":     true,
+	"qemu64":     true,
+}
+
+// getCPUInfo processes all cpu info data and returns
+// slice of usable cpu models and features.
+func (n *NodeLabeller) getCPUInfo() ([]string, map[string]bool) {
```

Comment:
The second return value contains CPU features. Does it make sense to use the `cpuFeatures` struct here?